spec/features/two_factor_authentication/sign_in_spec.rb

```diff
@@ -105,7 +105,10 @@
       expect(page).not_to have_css('.progress-steps')
     end
 
-    scenario 'user who enters OTP incorrectly 3 times is locked out for OTP validity period' do
+    scenario 'user who enters OTP incorrectly 3 times is locked out for OTP validity period', js: true do
+      allow(Figaro.env).to receive(:session_check_frequency).and_return('1')
+      allow(Figaro.env).to receive(:session_check_delay).and_return('2')
```

Comment:
These settings are needed to make sure the test fails if the `handle_second_factor_locked_user` method is not doing the right thing. I made sure the test failed with the previous code before fixing it. Without these overrides, the default settings would apply, which would take too long to trigger the session timeout calls. The fact that the session timeout JS was still active after the lockout is what caused the bug I am fixing now. See my commit message for an explanation of the bug.